sources/osgStats/Stats.js

```diff
@@ -0,0 +1,404 @@
+
+    this._labelMaxWidth = 0;
+    this._viewport = viewport;
+
+    this._node = undefined;
+    this._text = undefined;
+    this._offsetGroup = undefined;
+
+    this._counters = {};
+    this._groups = [];
+    this._updates = [];
+
+    this._x = 0;
+    this._y = 0;
+    this._width = 0;
+    this._height = 0;
+
+    this._lineFactor = 1.1;
+    this._characterDisplayHeight = 32.0;
+
+    this._displayFilter = [];
+
+    this._backgroundWidth = 0;
+    this._backgroundHeight = 0;
+
+    this._graphToDisplay = [];
+
+    this._init(viewport, options);
+};
+
+MACROUTILS.createPrototypeObject(Stats, {
+    getCounter: function(name) {
+        if (!this._counters[name]) {
+            this._counters[name] = new Counter({
+                caption: name,
+                average: true
+            });
+        }
+
+        return this._counters[name];
+    },
+    addConfig: function(config) {
+        for (var valueName in config.values) {
+            if (this._counters[valueName]) {
+                notify.warn('Counter ' + valueName + ' already exist, overring it');
+            }
+            var valueConfig = config.values[valueName];
+            var counter = new Counter(valueConfig);
+            this._counters[valueName] = counter;
+        }
+
+        if (config.groups) {
+            for (var i = 0; i < config.groups.length; i++) {
+                var groupConfig = config.groups[i];
+                var group = {
+                    name: groupConfig.name ? groupConfig.name : groupConfig.caption,
+                    caption: groupConfig.caption,
+                    values: groupConfig.values
+                };
+                this._groups.push(group);
+            }
+        }
+
+        if (config.update) {
+            this._updates.push(config.update);
+        }
+    },
+    update: function() {
+        for (var i = 0; i < this._updates.length; i++) {
+            this._updates[i](this);
+        }
+        this._dirtyValues = true;
+
+        if (this._checkViewportChanged()) this._dirtyCaptions = true;
+
+        if (this._dirtyCaptions) {
+            this._bufferStats.resetCaptions();
+            this._generateCaptions();
+            this._dirtyCaptions = false;
+        }
+
+        if (this._dirtyValues) {
+            this._bufferStats.resetValues();
+            this._generateValues();
+            this._dirtyValues = false;
+        }
+    },
+    getNode: function() {
+        return this._node;
+    },
+    setShowFilter: function(groupNamesArray) {
+        this._dirtyCaptions = true;
+        if (!groupNamesArray) {
+            this._displayFilter.length = 0;
+            return;
+        }
+        this._displayFilter = groupNamesArray.slice();
+    },
+    setFontSize: function(size) {
+        this._characterDisplayHeight = size;
+        this._dirtyCaptions = true;
+    },
+    _init: function(viewport, options) {
+        // 3D init
+        this._bufferStats = new BufferStats();
```

Comment:
Shouldn't it be declared in the constructor? For best practices if I recall well.